### PR TITLE
clarify mesh gateway docs use cases; include admin partition flow

### PR DESCRIPTION
**Pull request by trujillo-adam**
This PR adds cross-partition information to the Service Mesh > Gateways > Mesh Gateways documentation. It also reorganizes existing content to make the use-cases clearer. The WAN federation information is unchanged.

### CODE REVIEW

File: website/content/docs/connect/gateways/mesh-gateway/service-to-service-traffic-partitions.mdx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            destination_namespace = "default"
+            destination_type = "service"
+            destination_name = "billing"
+            local_bind_port = 9090
+         }
+      ]
+   }
+}
+```
+
```
Comment (karl-cardenas-coding) [inline website/content/docs/connect/gateways/mesh-gateway/service-to-service-traffic-partitions.mdx:129]: We should showcase how to do this with k8s annotations as well.  Below is an example NOT using trproxy.
postgress is the upstream service located in another partitions.  Double check me, but I belive the syntax is `service_name.[partition_name].[namespace]:portNumber

```
consul.hashicorp.com/connect-inject: "true"
consul.hashicorp.com/connect-service-upstreams: "postgres.default.default:5432"
```

**Review by karl-cardenas-coding — APPROVED** [review]
Looks good, some minor feedback

**Comment by trujillo-adam** [comment]
@karl-cardenas-coding - I added HCL examples using the HCL converter tool, but I didn't know how to implement this comment:https://github.com/hashicorp/consul/pull/11859#discussion_r773302234

We can open an issue to improve the examples later.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/535948.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit a08f294602d73d538a1df5cc75ea4e06c4d1ea5b onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit a08f294602d73d538a1df5cc75ea4e06c4d1ea5b onto `release/1.11.x` succeeded!

**Pull request merged**